torchprime/torch_xla_models/train.py

```diff
@@ -317,11 +323,27 @@ def step_closure(epoch, step, loss, trace_start_time, trace_end_time):
     xm.wait_device_ops()
     logger.info("Finished training run")
 
-    # Analyze the step duration from the latest profile
     if self.config.profile_step >= 0:
+      # Analyze the step duration from the latest profile
       step_duration = step_duration_from_latest_profile(self.config.profile_dir)
       metrics_logger.log_step_execution_time(step_duration)
 
+      # Compute the MFU.
+      tpu_name = os.environ.get("TORCHPRIME_TPU_TYPE", None)
+      self.config.model.torch_dtype = (
+        torch.get_default_dtype()
```

Comment:
Unfortunately this won't be accurate; we'll need to inspect the dtype from the model weights. The reason is that line 381 (https://github.com/AI-Hypercomputer/torchprime/pull/246/files#diff-6e7ee464035cac83b0c1171dbea2511919602ed800005c2aa355aef72d04c54cR381) is used to set the dtype, and that decorator only temporarily sets the dtype.

Here's one quick way to get the dtype from the model:

```python
def get_dtype(module):
  dtypes = {param.dtype for param in module.parameters()}
  if len(dtypes) != 1:
    raise ValueError(f"Inconsistent dtypes found: {dtypes}")
  return dtypes.pop()
```